core/trino-main/src/main/java/io/trino/sql/planner/optimizations/AddExchanges.java

```diff
@@ -246,13 +252,68 @@ public PlanWithProperties visitAggregation(AggregationNode node, PreferredProper
             }
             else if ((!isStreamPartitionedOn(child.getProperties(), partitioningRequirement) && !isNodePartitionedOn(child.getProperties(), partitioningRequirement)) ||
                     node.hasEmptyGroupingSet()) {
+                List<Symbol> partitioningKeys = parentPreferredProperties.getGlobalProperties()
+                        .flatMap(PreferredProperties.Global::getPartitioningProperties)
+                        .map(PreferredProperties.PartitioningProperties::getPartitioningColumns)
+                        .flatMap(partitioningColumns -> useParentPreferredPartitioning(node, partitioningColumns))
+                        .orElse(node.getGroupingKeys());
                 child = withDerivedProperties(
-                        partitionedExchange(idAllocator.getNextId(), REMOTE, child.getNode(), node.getGroupingKeys(), node.getHashSymbol()),
+                        partitionedExchange(idAllocator.getNextId(), REMOTE, child.getNode(), partitioningKeys, node.getHashSymbol()),
                         child.getProperties());
             }
             return rebaseAndDeriveProperties(node, child);
         }
 
+        /**
+         * For cases where parent prefers to be partitioned by a subset of the current node {@link AggregationNode#getGroupingKeys()},
+         * and the preferred partitioning columns have enough distinct values to provide good parallelism, we want to partition
+         * just by parent preferred partitioning columns to avoid another data shuffle required by the parent.
+         */
+        private Optional<List<Symbol>> useParentPreferredPartitioning(AggregationNode node, Set<Symbol> parentPreferredPartitioningColumns)
+        {
+            if (isUseExactPartitioning(session)) {
+                return Optional.empty();
+            }
+            if (parentPreferredPartitioningColumns.isEmpty()) {
+                return Optional.empty();
+            }
+            if (!ImmutableSet.copyOf(node.getGroupingKeys()).containsAll(parentPreferredPartitioningColumns)) {
+                // parent wants to be partitioned by at least one different column
+                return Optional.empty();
+            }
+            double parentPartitioningNDV = getMinDistinctValueCountEstimate(statsProvider.getStats(node), parentPreferredPartitioningColumns);
```

Comment:
What about test for this case?

Reply:
It's not that easy to do because `AddExchanes` is a `PlanOptimizer` not a Rule so test support for stats is limited but testing this makes sense so I will try to figure something out

Reply:
I added test for this by disabling  stats